data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/sink/KafkaSinkConfig.java

```diff
@@ -61,7 +67,7 @@ public void setDlqConfig(final PluginSetting pluginSetting) {
 
 
     @JsonProperty("topic")
-    TopicConfig topic;
+    SinkTopicConfig topic;
 
     @JsonProperty("authentication")
     private AuthConfig authConfig;
```

Comment:
I should have pointed this out previously. It looks like "AuthConfig" and "AwsConfig" and "bootstrapServers", "SchemaConfig" are common between "KafkaSourceConfig" and "KafkaSinkConfig", does it make sense to move put these in a common interface and a common base class?

Reply:
Same with getThreadWaitTime()/getThreadWaitingTime()

Reply:
I think that is a good idea, but probably beyond the scope of this PR. The goal of this PR is to decouple the topic configurations as they do need to vary between the different plugins.